test/utils/SafeCast.test.js

```diff
@@ -0,0 +1,80 @@
+const SafeCastMock = artifacts.require('SafeCastMock');
+const { expectRevert } = require('openzeppelin-test-helpers');
+
+contract('SafeCast', async (accounts) => {
+  describe('toUint128()', () => {
+    it('reverts on overflow (i.e. input >= 2^128)', async () => {
+      const contract = await SafeCastMock.new();
+      await expectRevert(
+        contract.toUint128('340282366920938463463374607431768211456'),
```

Comment:
Evaluating correctness of these tests should be easy: this could be achieved by using the `BN` type (included in the test helpers) and dynamically creating the test values (i.e. 2**n, 2**n + 1, 2**n -1). 

The equality checks should also be performed as
```javacript
expect(await contract.toUint128(0)).to.be.bignumber.equal(expected)
```

Reply:
We may even want to automatically generate the `describe` blocks from an array of word sizes (128, 64, 32, 16, 8), to prevent copy-paste errors, since all tests will look the same.

Reply:
Yes, I agree with this. Will have to spend a bit more time on this suggestion as JS is not my mother tongue. Will handle all of the other issues in one commit and then focus on these proposals! Thanks for your input here.